Libs/MRML/Core/vtkMRMLPlotNode.h

```diff
@@ -0,0 +1,249 @@
+/*==============================================================================
+
+  Copyright (c) Kapteyn Astronomical Institute
+  University of Groningen, Groningen, Netherlands. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Davide Punzo, Kapteyn Astronomical Institute,
+  and was supported through the European Research Consil grant nr. 291531.
+
+==============================================================================*/
+
+#ifndef __vtkMRMLPlotNode_h
+#define __vtkMRMLPlotNode_h
+
+#include <string>
+#include <vector>
+
+// MRML Includes
+#include "vtkMRMLNode.h"
+
+class vtkMRMLStorageNode;
+class vtkMRMLTableNode;
+
+// VTK Includes
+class vtkColor4ub;
+class vtkPlot;
+class vtkTable;
+
+/// \brief MRML node to represent a Plot object
```

Comment:
This seems to be unnecessarily complicated. A MRML node should be a higher-level concept that a user can visualize, edit, etc. A vtkPlot object is lower level: you cannot display a vtkPlot object - you need a vtkChart (and scene, etc) for that.

So, instead of having a separate vtkMRMLPlotNode and vtkMRMLPlotLayoutNode, I would recommend to store all data in a single vtkMRMLPlotNode class. The implementation would be simpler than the current one, because you would not need to manage so many node references, you would store all the data in a single class (you would still need to use node references for table nodes, but not for plot nodes).

You can have a look at how the [sequence browser node](https://github.com/SlicerRt/Sequences/blob/master/SequenceBrowser/MRML/vtkMRMLSequenceBrowserNode.cxx) refers to multiple nodes and associated metadata for each node: It stores a list of "postfix" strings (usually "0", "1", "2", ...) uses that to generate unique role names and unique attribute names.

```
<SequenceBrowser id="vtkMRMLSequenceBrowserNode1" name="SequenceBrowser"
 virtualNodePostfixes="0 1 2" references="dataNodeRef0:vtkMRMLMarkupsFiducialNode1;dataNodeRef1:vtkMRMLMarkupsFiducialNode2;dataNodeRef2:vtkMRMLMarkupsFiducialNode3;sequenceNodeRef0:vtkMRMLSequenceNode2;sequenceNodeRef1:vtkMRMLSequenceNode3;sequenceNodeRef2:vtkMRMLSequenceNode4;"
 SynchronizationPropertiesMap0="playback true recording true overwriteProxyName false saveChanges false "  
 SynchronizationPropertiesMap1="playback true recording true overwriteProxyName true saveChanges true "  
 SynchronizationPropertiesMap2="playback false recording false overwriteProxyName false saveChanges false " >
</SequenceBrowser>
```
For plots, it would look something like this:
```
<Plot id="vtkMRMLSequenceBrowserNode1" name="SequenceBrowser"
 ...
 plotPostfixes="0 1 3"
 references="tableNodeRef0:vtkMRMLTableNode1;tableNodeRef1:vtkMRMLTableNode2;tableNodeRef3:vtkMRMLTableNode5"
 plotProperties0="name:My plot 1|type:line|xColumnName:time|yColumnName:distance"
 plotProperties1="name:My plot 2|type:line|xColumnName:time|yColumnName:distance"
 plotProperties3="name:My plot 3|type:line|xColumnName:time|yColumnName:distance"
</Plot>
```